Specs/Renderer/BufferSpec.js

```diff
@@ -11,22 +11,21 @@ defineSuite([
         createContext) {
     'use strict';
 
-    createBufferSpecs();
-
-    var c = createContext({ requestWebgl2 : true });
-    if (c.webgl2) {
-        createBufferSpecs({ requestWebgl2 : true });
-    }
-    c.destroyForSpecs();
+    createBufferSpecs({});
+    createBufferSpecs({
+        requestWebgl2 : true
+    });
 
     function createBufferSpecs(contextOptions) {
-        var context;
         var buffer;
         var buffer2;
+        var context = createContext(contextOptions);
```

Comment:
Right, but it creates those contexts when it queues the tests. The contexts are created but the tests aren't run immediately. Any number of tests can run before these tests, many of which will create an destroy contexts. If too many are created before these tests are run, these two contexts may be lost. That's why we always create a context needed for the tests in `beforeAll` or `beforeEach` function.